### DISCUSSION TITLE
What is the best practice to modify AST node and generate the code?

### DISCUSSION THREAD

**Discussion by hyphappy011** [eclipse-langium/langium#1354] [Q&A]
hello,
I had designed a tool to drag and drop flow chart, which is driven by json。the user drag and drop some nodes, and generate json code, and then the web page render flow graph。
now i want convert the json to langium language AST, and then convert the AST to customer DSL(defined by langium)。of course, user can writer the code, generate AST automatic，and convert AST json the render graph Code.
Is there any suggestion or example for me?
<img width="1049" alt="image" src="https://github.com/eclipse-langium/langium/assets/24590408/2777b1f0-372b-40e2-8b22-3213f042702c">
<img width="1422" alt="image" src="https://github.com/eclipse-langium/langium/assets/24590408/57c77942-b7c1-4047-9f71-95f0bc7bc4d3">

**Comment by msujew** [comment]
Hey @hyphappy011,

general string serialization (i.e. converting any AST-like representation back to DSL code) is still on our todo-list, so there's no automatic way of doing that yet. You will likely need to implement a custom code generator for this.

The best solution would only perform delta updates, i.e. identify which elements you need to insert/change and only apply those text changes to the document. That way you don't accidentally delete hidden terminals like comments.
